inc/class-upgrade.php

```diff
@@ -608,6 +612,19 @@ private function upgrade_772() {
 		}
 	}
 
+	/**
+	 * Performs the 9.0 upgrade.
+	 *
+	 * @return void
+	 */
+	private function upgrade90() {
+		// Invalidate all sitemap cache transients.
+		WPSEO_Sitemaps_Cache_Validator::invalidate_storage();
```

Comment:
Have you considered using `WPSEO_Sitemaps_Cache_Validator::cleanup_database();` ?

This will also remove all present transients for sites that did not have Object Cache enabled before but have stored entries in the options table.